back-end/controllers/UserController.js

```diff
@@ -17,8 +18,18 @@ const getUser = async (req, res) => {
 const updateUserById = async (req, res) => {
   try {
     const { id } = req.params;
+    const existEmail = await findByEmail(req.body.email);
+    if (existEmail) {
+      const body = {
+        fullName: req.body.fullName,
+        dateOfBirth: req.body.dateOfBirth,
+      };
+      await updateByID(id, body);
+      return res.status(400).json({ message: 'exist email' });
+    }
     const user = await updateByID(id, req.body);
-    res.status(200).json(user);
+    const token = jwt.sign({ email: user.email }, process.env.SECRET_KEY, { expiresIn: '2h' });
```

Comment:
We have the same code in `authService` when we do login, so you can create a method with this logic.